builtin/tests/batch_onboarding_deals.rs

```diff
@@ -0,0 +1,271 @@
+    let mut market_state: fil_actor_market::State =
+        w.find_actor_state(STORAGE_MARKET_ACTOR_ID).unwrap().unwrap();
+    let deal_keys: Vec<BytesKey> = deals.iter().map(|(id, _)| deal_id_key(*id)).collect();
+    let alloc_ids = market_state
+        .get_pending_deal_allocation_ids(&DynBlockstore::new(w.store()), &deal_keys)
+        .unwrap();
+    assert_eq!(BATCH_SIZE, alloc_ids.len());
+
+    println!("alloc_ids: {:?}", alloc_ids);
+
+    // Associate deals with sectors.
+    let sector_precommit_data = deals
+        .into_iter()
+        .map(|(id, _deal)| PrecommitMetadata {
+            deals: vec![id],
+            commd: CompactCommD(Some(make_piece_cid(b"test data"))),
+        })
+        .collect();
+
+    // Pre-commit as single batch.
+    let precommits = precommit_sectors_v2(
+        &mut w,
+        BATCH_SIZE,
+        BATCH_SIZE,
+        sector_precommit_data,
+        &worker.id_addr(),
+        &miner,
+        SEAL_PROOF,
+        0,
+        true,
+        Some(sector_duration),
+        PRECOMMIT_V2,
+    );
+    let first_sector_no = precommits[0].info.sector_number;
+
+    // Prove-commit as a single aggregate.
+    w.set_epoch(w.epoch() + PRE_COMMIT_CHALLENGE_DELAY + 1);
+
+    println!("================================= Running PCA ===============================");
+    let pca_result = prove_commit_aggregate(&mut w, &worker.id_addr(), &miner, precommits);
+
+    println!("=================================== PCA Trace ===============================");
+    let analysis = TraceAnalysis::build(pca_result.trace);
+    println!("{}", analysis.format_spans());
+
+    // Submit Window PoST to activate power.
+    let (dline_info, p_idx) = advance_to_proving_deadline(&mut w, &miner, 0);
+
+    let sector_size = SEAL_PROOF.sector_size().unwrap();
+    let st: MinerState = w.find_actor_state(miner_id).unwrap().unwrap();
+    let sector = st.get_sector(&DynBlockstore::new(w.store()), first_sector_no).unwrap().unwrap();
+    let mut expect_new_power = power_for_sector(sector_size, &sector);
+    // Confirm the verified deal resulted in QA power.
+    assert_eq!(&expect_new_power.raw * 10, expect_new_power.qa);
+    expect_new_power.raw *= BATCH_SIZE;
+    expect_new_power.qa *= BATCH_SIZE;
+    submit_windowed_post(
+        &mut w,
+        &worker.id_addr(),
+        &miner,
+        dline_info,
+        p_idx,
+        Some(expect_new_power.clone()),
+    );
+
+    // Verify state expectations.
+    let balances = get_miner_balance(&mut w, miner_id);
+    assert!(balances.initial_pledge.is_positive());
+
+    // let network_stats = get_network_stats(v);
+    // assert_eq!(
+    //     network_stats.total_bytes_committed,
+    //     BigInt::from(sector_size as usize * BATCH_SIZE)
+    // );
+    // assert_eq!(network_stats.total_qa_bytes_committed, network_stats.total_bytes_committed * 10);
+    // assert!(network_stats.total_pledge_collateral.is_positive());
+}
+
+// TODO use deal batcher instead
```

Comment:
I've updated this in builtin-actors. Fine to use this version here, but remove the TODOs